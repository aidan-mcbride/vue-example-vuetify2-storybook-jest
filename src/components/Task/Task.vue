<template>
  <v-list-item>
    <v-list-item-action>
      <v-btn
        class="task__archive-button"
        aria-label="archive task"
        text
        icon
        @click="$emit('archiveTask', task.id)"
      >
        <v-icon>{{ checkboxIcon }}</v-icon>
      </v-btn>
    </v-list-item-action>
    <v-list-item-content>
      <v-list-item-title>{{ task.title }}</v-list-item-title>
    </v-list-item-content>
    <v-list-item-action>
      <v-btn
        v-if="!isChecked"
        class="task__pin-button"
        aria-label="pin task"
        text
        icon
        @click="$emit('pinTask', task.id)"
      >
        <v-icon v-if="isPinned" color="blue">mdi-star</v-icon>
        <v-icon v-else>mdi-star-outline</v-icon>
      </v-btn>
    </v-list-item-action>
  </v-list-item>
</template>

<script>
export default {
  name: 'Task',
  props: {
    task: {
      type: Object,
      required: true,
      default: () => ({
        id: '',
        state: '',
        title: ''
      })
    }
  },
  computed: {
    isChecked() {
      return this.task.state === 'TASK_ARCHIVED';
    },
    isPinned() {
      return this.task.state === 'TASK_PINNED';
    },
    checkboxIcon() {
      return this.isChecked
        ? 'mdi-checkbox-marked-outline'
        : 'mdi-checkbox-blank-outline';
    }
  }
};
</script>
