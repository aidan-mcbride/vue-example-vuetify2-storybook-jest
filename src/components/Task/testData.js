export const taskData = {
  id: '1',
  title: 'Test Task',
  state: 'TASK_INBOX',
  updated_at: new Date(2019, 0, 1, 9, 0)
};
