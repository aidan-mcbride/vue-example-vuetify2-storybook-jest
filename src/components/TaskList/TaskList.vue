<template>
  <pure-task-list
    :tasks="tasks"
    @archiveTask="archiveTask"
    @pinTask="pinTask"
  />
</template>

<script>
import PureTaskList from './PureTaskList';
import { mapState, mapActions } from 'vuex';

export default {
  name: 'TaskList',
  components: {
    PureTaskList
  },
  computed: {
    ...mapState(['tasks'])
  },
  methods: {
    ...mapActions(['archiveTask', 'pinTask'])
  }
};
</script>
