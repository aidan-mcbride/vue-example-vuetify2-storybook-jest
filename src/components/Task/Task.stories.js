import { action } from '@storybook/addon-actions';
import Task from './Task';

import { taskData } from './testData';
export default {
  title: 'Task',
  // Our exports that end in "Data" are not stories.
  excludeStories: /.*Data$/
};
export const actionsData = {
  onPinTask: action('onPinTask'),
  onArchiveTask: action('onArchiveTask')
};

const taskTemplate = `<task :task="task" @archiveTask="onArchiveTask" @pinTask="onPinTask"/>`;

// default task state
export const Default = () => ({
  components: { Task },
  template: taskTemplate,
  props: {
    task: {
      default: () => taskData
    }
  },
  methods: actionsData
});
// pinned task state
export const Pinned = () => ({
  components: { Task },
  template: taskTemplate,
  props: {
    task: {
      default: () => ({
        ...taskData,
        state: 'TASK_PINNED'
      })
    }
  },
  methods: actionsData
});
// archived task state
export const Archived = () => ({
  components: { Task },
  template: taskTemplate,
  props: {
    task: {
      default: {
        ...taskData,
        state: 'TASK_ARCHIVED'
      }
    }
  },
  methods: actionsData
});
