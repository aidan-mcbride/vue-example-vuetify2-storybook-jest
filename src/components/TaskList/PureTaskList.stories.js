import PureTaskList from './PureTaskList';

import { defaultTasksData, withPinnedTasksData } from './testData';
import { actionsData } from '@/components/Task/Task.stories';

const paddedList = () => {
  return {
    template:
      '<div style="padding: 3rem; background-color: #CFD8DC"><story/></div>'
  };
};

export default {
  title: 'TaskList',
  excludeStories: /.*Data$/,
  decorators: [paddedList]
};

// default TaskList state
export const Default = () => ({
  components: { PureTaskList },
  template: `<pure-task-list :tasks="tasks" @archiveTask="onArchiveTask" @pinTask="onPinTask"/>`,
  props: {
    tasks: {
      default: () => defaultTasksData
    }
  },
  methods: actionsData
});

// TaskList with pinned tasks
export const WithPinnedTasks = () => ({
  components: { PureTaskList },
  template: `<pure-task-list :tasks="tasks" @archiveTask="onArchiveTask" @pinTask="onPinTask"/>`,
  props: {
    tasks: {
      default: () => withPinnedTasksData
    }
  },
  methods: actionsData
});

// loading TaskList state
export const Loading = () => ({
  components: { PureTaskList },
  template: `<pure-task-list loading @archiveTask="onArchiveTask" @pinTask="onPinTask"/>`,
  methods: actionsData
});

// empty TaskList state
export const Empty = () => ({
  components: { PureTaskList },
  template: `<pure-task-list @archiveTask="onArchiveTask" @pinTask="onPinTask"/>`,
  methods: actionsData
});
