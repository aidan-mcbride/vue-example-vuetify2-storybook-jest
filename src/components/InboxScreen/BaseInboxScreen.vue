<template>
  <v-container fluid>
    <v-alert v-if="error" type="error" prominent>
      <div class="title">Oh No!</div>
      <div>Something went wrong</div>
    </v-alert>

    <div v-else>
      <nav>
        <h1 class="title primary--text">Taskbox</h1>
      </nav>

      <task-list />
    </div>
  </v-container>
</template>

<script>
import TaskList from '@/components/TaskList';

export default {
  name: 'PureInboxScreen',
  components: {
    TaskList
  },
  props: {
    error: {
      type: Boolean,
      default: false
    }
  }
};
</script>
