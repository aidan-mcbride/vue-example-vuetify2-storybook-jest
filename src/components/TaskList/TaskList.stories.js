import TaskList from './TaskList.vue';

import { defaultTasksData } from './testData';
import { actionsData } from '@/components/Task/Task.stories';

const paddedList = () => {
  return {
    template: '<div style="padding: 3rem;"><story/></div>'
  };
};

export default {
  title: 'TaskList',
  excludeStories: /.*Data$/,
  decorators: [paddedList]
};

// default TaskList state
export const Default = () => ({
  components: { TaskList },
  template: `<task-list :tasks="tasks" @archiveTask="onArchiveTask" @pinTask="onPinTask"/>`,
  props: {
    tasks: {
      default: () => defaultTasksData
    }
  },
  methods: actionsData
});
