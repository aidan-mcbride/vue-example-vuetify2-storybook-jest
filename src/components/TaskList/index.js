import TaskList from './TaskList.vue';
import BaseTaskList from './BaseTaskList.vue';

export { TaskList, BaseTaskList };
