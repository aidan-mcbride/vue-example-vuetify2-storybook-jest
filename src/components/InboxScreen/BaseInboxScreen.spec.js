import Vuex from 'vuex';

// Components
import BaseInboxScreen from './BaseInboxScreen';
import Task from '@/components/Task';

// Utilities
import { mount, createLocalVue } from '@vue/test-utils';
import { defaultTasksData } from '../TaskList/testData';

const localVue = createLocalVue();
localVue.use(Vuex);

const store = new Vuex.Store({
  state: {
    tasks: [...defaultTasksData]
  }
});

describe('BaseInboxScreen', () => {
  describe('Default state', () => {
    it('Renders with tasks from store', () => {
      // arrange
      const wrapper = mount(BaseInboxScreen, {
        store,
        localVue
      });
      const task = wrapper.find(Task);
      const firstTaskTitle = defaultTasksData[0].title;
      // assert
      expect(wrapper.html()).toMatchSnapshot();
      expect(task.text()).toContain(firstTaskTitle);
    });
  });

  describe('Error state', () => {
    it('Renders correct markup', () => {
      // arrange
      const wrapper = mount(BaseInboxScreen, {
        propsData: {
          error: true
        }
      });
      // assert
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
