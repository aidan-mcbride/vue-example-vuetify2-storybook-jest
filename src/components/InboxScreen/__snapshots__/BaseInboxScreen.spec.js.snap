// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseInboxScreen Default state Renders with tasks from store 1`] = `
<div class="container container--fluid">
  <div>
    <nav>
      <h1 class="title">Taskbox</h1>
    </nav>
    <div>
      <!---->
      <!---->
      <div role="list" class="v-list v-sheet v-sheet--tile theme--light">
        <div tabindex="-1" role="listitem" class="v-list-item theme--light">
          <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
          <div class="v-list-item__content">
            <div class="v-list-item__title">Task 1</div>
          </div>
          <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star-outline theme--light"></i></span></button></div>
        </div>
        <div tabindex="-1" role="listitem" class="v-list-item theme--light">
          <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
          <div class="v-list-item__content">
            <div class="v-list-item__title">Task 2</div>
          </div>
          <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star-outline theme--light"></i></span></button></div>
        </div>
        <div tabindex="-1" role="listitem" class="v-list-item theme--light">
          <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
          <div class="v-list-item__content">
            <div class="v-list-item__title">Task 3</div>
          </div>
          <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star-outline theme--light"></i></span></button></div>
        </div>
        <div tabindex="-1" role="listitem" class="v-list-item theme--light">
          <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
          <div class="v-list-item__content">
            <div class="v-list-item__title">Task 4</div>
          </div>
          <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star-outline theme--light"></i></span></button></div>
        </div>
        <div tabindex="-1" role="listitem" class="v-list-item theme--light">
          <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
          <div class="v-list-item__content">
            <div class="v-list-item__title">Task 5</div>
          </div>
          <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star-outline theme--light"></i></span></button></div>
        </div>
        <div tabindex="-1" role="listitem" class="v-list-item theme--light">
          <div class="v-list-item__action"><button type="button" class="task__archive-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="archive task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i></span></button></div>
          <div class="v-list-item__content">
            <div class="v-list-item__title">Task 6</div>
          </div>
          <div class="v-list-item__action"><button type="button" class="task__pin-button v-btn v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="pin task"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-star-outline theme--light"></i></span></button></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`BaseInboxScreen Error state Renders correct markup 1`] = `
<div class="container container--fluid">
  <div role="alert" class="v-alert v-sheet theme--dark v-alert--prominent error">
    <div class="v-alert__wrapper"><i aria-hidden="true" class="v-icon notranslate v-alert__icon material-icons theme--dark">$error</i>
      <div class="v-alert__content">
        <div class="title">Oh No!</div>
        <div>Something went wrong</div>
      </div>
    </div>
  </div>
</div>
`;
